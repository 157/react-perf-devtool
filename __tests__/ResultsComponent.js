var React = require('react')
var ReactTestRenderer = require('react-test-renderer')

var Results = require('../src/extension/components/Results')

var rawMeasures = require('../samples/measures')

describe('Results component', () => {
  it('should output the calculated results', () => {
    var tree = ReactTestRenderer.create(
      <Results
        rawMeasures={rawMeasures}
        totalTime={124.45}
        loading={false} // Consider the queue is empty and the task is finished (event loop)
      />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
