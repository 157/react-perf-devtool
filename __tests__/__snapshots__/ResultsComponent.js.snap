// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results component should output the calculated results 1`] = `
<div>
  <div
    style={
      Object {
        "padding": "8px",
      }
    }
  >
    <p>
      Time taken by all the components:
       
      <strong>
        124.45
         ms
      </strong>
    </p>
    <p>
      Committing changes took:
       
      <strong>
        7.35
         ms
      </strong>
    </p>
    <p>
      Committing
       
      <strong>
        8
      </strong>
       host effects took:
       
      <strong>
        1.23
         ms
      </strong>
    </p>
    <p>
      Calling
       
      <strong>
        7
      </strong>
       
      lifecycle methods took:
       
      <strong>
        5.27
         ms
      </strong>
    </p>
    <p>
      Total time:
       
      <strong>
        138.30
         
        ms
      </strong>
    </p>
  </div>
</div>
`;
